monitoring/metrics.rb

```diff
@@ -0,0 +1,250 @@
+require "google/cloud/monitoring"
+
+def create_metric_descriptor project_id:
+  # [START monitoring_create_metric]
+  client = Google::Cloud::Monitoring::Metric.new
+  project_name = Google::Cloud::Monitoring::V3::MetricServiceClient.project_path project_id
+
+  # Random suffix for metric type to avoid collisions with other runs
+  random_suffix = rand(36**10).to_s(36)
```

Comment:
Using a random name to avoid test collisions is necessary.

But the random-name generation code shouldn't be included in the sample code.  In other words, it should live outside the [START yadayada] and [END yadayada] doc tags.  For example, https://github.com/GoogleCloudPlatform/ruby-docs-samples/blob/master/spanner/spec/spanner_samples_spec.rb (the test code) creates a random name.

Reply:
ok moving outside of the START - END block